src/Symfony/Bridge/Twig/Extension/DumpExtension.php

```diff
@@ -67,11 +70,14 @@ public function dump(\Twig_Environment $env, $context)
         }
 
         $dump = fopen('php://memory', 'r+b');
-        $dumper = new HtmlDumper($dump);
+        $prevOutput = $this->dumper->setOutput($dump);
```

Comment:
Instead of setOutput, you could use dump's 2nd arg.
An you could also create a new interface to enforce what you need here.